### PR TITLE
Migration script fixes from 1.2.0.2

### PR DESCRIPTION
**Pull request by ase-101**

### CODE REVIEW

**Review by hemant-vadher — APPROVED** [review]

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql [hunk 1]
```diff
@@ -245,6 +247,9 @@ ALTER TABLE master.batch_job_execution_params ALTER COLUMN string_val TYPE varch
 ALTER TABLE master.blocklisted_words DROP CONSTRAINT IF EXISTS pk_blwrd_code;
 ALTER TABLE master.blocklisted_words ALTER COLUMN lang_code DROP NOT NULL;
 ALTER TABLE master.blocklisted_words ADD CONSTRAINT pk_blwrd_code PRIMARY KEY (word);
+
+INSERT into master.blocklisted_words (word,descr,lang_code,is_active,cr_by,cr_dtimes,upd_by,upd_dtimes,is_deleted,del_dtimes) SELECT word,descr,lang_code,is_active,cr_by,cr_dtimes,upd_by,upd_dtimes,is_deleted,del_dtimes FROM master.blacklisted_words;
```
Comment (vishwa-vyom) [inline db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql:251]: As per this we can see that old table is still kept, rather this could have been a simple table rename. If changed, please take care of the same in the revoke script

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql [hunk 2]
```diff
@@ -255,6 +260,16 @@ ALTER TABLE master.user_detail_h DROP COLUMN uin;
 ALTER TABLE master.user_detail_h DROP COLUMN email;
 ALTER TABLE master.user_detail_h DROP COLUMN mobile;
 
+SELECT * INTO master.template_copy FROM master.template;
```
Comment (vishwa-vyom) [inline db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql:263]: If for backup, please use the bkp convention to be consistent everywhere

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql [hunk 3]
```diff
@@ -255,6 +260,16 @@ ALTER TABLE master.user_detail_h DROP COLUMN uin;
...
+
+UPDATE master.template set module_id='10001' where module_id='10002' and template_typ_code not like 'reg-%';
```
Comment (vishwa-vyom) [inline db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql:265]: Please add a detailed comment on why we are doing this module id changes, since this is not directly related to migration but related to performance or cleanup. Lets include if there is any warning also.

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql (hunk 2 above)
Comment (vishwa-vyom) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql:263]: take care of this and below in the revoke script also

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: db_release_scripts/mosip_master/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+Directory contains DDL scripts for tables to be created under mosip_master DB.
+
+## dml
+
+Directory contains DML scripts for tables to be created under mosip_master DB.
+
+## migration_scripts
+
+Directory contains python scripts to be executed before / after DB migrations. Python scripts are stored under sub directory named after the migrated version.
+
```
[the remark this answers is not in the document]
Comment (ase-101) [reply db_release_scripts/mosip_master/README.md:12]: Procedure added in respective version folder.

File: db_release_scripts/mosip_master/migration_scripts/1.2.0/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+## migration-dynamicfield.py
+
+This script should be executed after DB upgrade.
+
```
[the remark this answers is not in the document]
Comment (ase-101) [reply db_release_scripts/mosip_master/migration_scripts/1.2.0/README.md:4]: Added details on execution and required permissions.

File: db_release_scripts/mosip_master/migration_scripts/1.2.0/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+## UI Spec migration
+
+TODO
+
+Note: Pre-Registration UI-spec was part of mosip-config and should be manually published using masterdata UI-spec API.
```
[the remark this answers is not in the document]
Comment (ase-101) [reply db_release_scripts/mosip_master/migration_scripts/1.2.0/README.md]: Done

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql (hunk 1 above)
Comment (ase-101) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql]: Done

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql (hunk 2 above)
Comment (ase-101) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql]: Done

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql (hunk 3 above)
Comment (ase-101) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql]: Done

Comment (vishwa-vyom) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql]: Resolved

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql (hunk 2 above)
Comment (vishwa-vyom) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql]: In one way it make sense, but now admin will not be consistant with all the other modules. Also in some cases, the full db backup will take a very long time and there might be large table which does not have any change.

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql (hunk 1 above)
Comment (vishwa-vyom) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql]: Resolved

File: db_release_scripts/mosip_master/migration_scripts/1.2.0/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply db_release_scripts/mosip_master/migration_scripts/1.2.0/README.md]: Resolved

File: db_release_scripts/mosip_master/migration_scripts/1.2.0/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply db_release_scripts/mosip_master/migration_scripts/1.2.0/README.md:4]: Resolved

File: db_release_scripts/mosip_master/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply db_release_scripts/mosip_master/README.md:12]: Resolved

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql [hunk 7]
```diff
@@ -23,133 +23,69 @@
...
+UPDATE master.template set module_id='10002' where template_typ_code like 'reg-dashboard%';
+DROP TABLE IF EXISTS master.template_migr_bkp;
+
+--------------------------------------------------------------------------------------------------------------------
+ALTER TABLE master.blacklisted_words DROP CONSTRAINT IF EXISTS pk_blwrd_code CASCADE;
+\ir ../ddl/master-blocklisted_words.sql
+ALTER TABLE master.blocklisted_words DROP CONSTRAINT IF EXISTS pk_blwrd_code CASCADE;
+ALTER TABLE master.blocklisted_words ALTER COLUMN lang_code DROP NOT NULL;
+INSERT into master.blocklisted_words (word,descr,lang_code,is_active,cr_by,cr_dtimes,upd_by,upd_dtimes,is_deleted,del_dtimes) SELECT distinct word,descr,lang_code,is_active,cr_by,cr_dtimes,upd_by,upd_dtimes,is_deleted,del_dtimes FROM master.blacklisted_words;
+DELETE FROM master.blocklisted_words where lang_code != :primary_language_code;
```
Comment (vishwa-vyom) [inline db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql:53]: blocklisted_words of non primary language should also be kept

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql (hunk 2 above)
Comment (vishwa-vyom) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql]: Lets remove the whole db backup point from documentation

**Review by vishwa-vyom — APPROVED** [review]

File: db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql (hunk 2 above)
Comment (ase-101) [reply db_release_scripts/mosip_master/sql/1.2.0_master-scripts_release.sql]: updated the document about revoke script and also about the backups taken.

**Review by vishwa-vyom — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]
